src/field.cpp

```diff
@@ -461,6 +461,30 @@ field_id field_from_ident(const std::string &field_ident)
     return fd_null;
 }
 
+void map::create_burnproducts( const tripoint p, item &fuel ) {
```

Comment:
Does the function really need a reference to a *non*-const `item`?

Reply:
Allowing for materials to yield different burn by-products. Main use would be for items that should melt rather than burn. Future use might be for tin and other metals that fire could melt.